### PR TITLE
Update cross-custodial - removed CONNECTED from PublisherStatus

### PR DESCRIPTION
**Pull request by sujitacharya2005**
Resolves  https://github.com/brave/brave-browser/issues/27255

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
image

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/BraveRewardsPublisher.java [hunk 1]
```diff
@@ -13,10 +13,9 @@ public class BraveRewardsPublisher {
     //PublisherStatus @
     //vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom
     @Retention(RetentionPolicy.SOURCE)
-    @IntDef({NOT_VERIFIED, CONNECTED, UPHOLD_VERIFIED, BITFLYER_VERIFIED, GEMINI_VERIFIED})
+    @IntDef({NOT_VERIFIED, UPHOLD_VERIFIED, BITFLYER_VERIFIED, GEMINI_VERIFIED})
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/BraveRewardsPublisher.java:16]: we should remove this IntDef altogether. instead we should use mojom struct for the same.

**Review by deeppandya — CHANGES_REQUESTED** [review]

File: android/java/org/chromium/chrome/browser/BraveRewardsNativeWorker.java [hunk 2]
```diff
@@ -587,11 +587,10 @@ private void setNativePtr(long nativePtr) {
 
     @CalledByNative
     public void OnPublisherInfo(int tabId) {
-        @PublisherStatus int pubStatus = GetPublisherStatus(tabId);
-        boolean verified = (pubStatus == BraveRewardsPublisher.CONNECTED
-                                   || pubStatus == BraveRewardsPublisher.UPHOLD_VERIFIED
-                                   || pubStatus == BraveRewardsPublisher.BITFLYER_VERIFIED
-                                   || pubStatus == BraveRewardsPublisher.GEMINI_VERIFIED)
+        int pubStatus = GetPublisherStatus(tabId);
+        boolean verified = (pubStatus == PublisherStatus.UPHOLD_VERIFIED
```
Comment (szilardszaloki) [inline android/java/org/chromium/chrome/browser/BraveRewardsNativeWorker.java:591]: For short: `boolean verified = pubStatus != PublisherStatus.NOT_VERIFIED;`. 🙂

Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/BraveRewardsNativeWorker.java]: updated all this condition other places

File: android/java/org/chromium/chrome/browser/rewards/BraveRewardsCreatorPanelFragment.java [hunk 3]
```diff
@@ -276,37 +271,34 @@ public void OnGetExternalWallet(String externalWallet) {
...
-    private void setPublisherNoteText(@PublisherStatus int pubStatus, int walletStatus) {
+    private void setPublisherNoteText(int pubStatus, int walletStatus) {
         String notePart1 = "";
         String walletType = BraveRewardsNativeWorker.getInstance().getExternalWalletType();
         if (walletStatus == WalletStatus.NOT_CONNECTED) {
-            if (pubStatus == BraveRewardsPublisher.CONNECTED
-                    || pubStatus == BraveRewardsPublisher.UPHOLD_VERIFIED
-                    || pubStatus == BraveRewardsPublisher.BITFLYER_VERIFIED
-                    || pubStatus == BraveRewardsPublisher.GEMINI_VERIFIED) {
+            if (pubStatus == PublisherStatus.UPHOLD_VERIFIED
```
Comment (szilardszaloki) [inline android/java/org/chromium/chrome/browser/rewards/BraveRewardsCreatorPanelFragment.java:283]: `pubStatus != PublisherStatus.NOT_VERIFIED`

**Review by szilardszaloki — APPROVED** [review]
LGTM, but let's see what @deeppandya has to say about it.

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by samartnik — APPROVED** [review]
lgtm

**Review by deeppandya — APPROVED** [review]
LGTM

**Pull request merged**